### PR TITLE
Move process metric attributes to the registry

### LINKED ISSUE #650: Process metrics attributes moved to registry
It sounds like the new norm is for all attributes to be added to the registry even if they are only used in one namespace. The `process.cpu.state` metric needs to be moved to the registry and changed to a `ref` in the process metrics yaml spec.

Discussed in #330

### PR DESCRIPTION
**Pull request by braydonk**
Fixes #650

## Changes

The guidance is for all attributes to be moved to global registry instead of being in metric specification files. This PR moves all attributes from `process.yaml` spec to the Process registry.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [N/A] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: CHANGELOG.md
Comment (dmitryax) [inline CHANGELOG.md:33]: Not sure if it should go to the user-facing changelog. I think we should ignore the changelog for this type of changes when no-op for the end users

Comment (braydonk) [reply CHANGELOG.md]: That makes sense. Removed in [b72b784](https://github.com/open-telemetry/semantic-conventions/pull/681/commits/b72b7845a20b197cfd104cb4ba4dad3b70394329)

File: model/registry/process.yaml
Comment (joaopgrassi) [inline model/registry/process.yaml:79]: Why isn't this nested together with the rest of attributes here? At the top there's already 

```
  - id: registry.process
    prefix: process
```

You can define `cpu.state` already below the others, no?

File: model/registry/process.yaml
Comment (joaopgrassi) [inline model/registry/process.yaml:83]: Since this is in the registry, it's not about only metrics anymore. Either remove this or maybe change the sentence so it's "generic"

File: model/registry/process.yaml
Comment (joaopgrassi) [inline model/registry/process.yaml:86]: Also here: It talks about metric things (data points) but we are in the registry. This should be adapted when you use the attribute `ref` in the process-metrics yaml file.

**Review by joaopgrassi — CHANGES_REQUESTED** [review]

File: model/registry/process.yaml
Comment (braydonk) [reply model/registry/process.yaml:79]: The above group has type `resource`, so I thought I had to make a separate group for `attribute_group`. Is that not the case?

File: model/registry/process.yaml
Comment (braydonk) [reply model/registry/process.yaml:83]: I don't see these attributes being used on anything other than process metrics, so I'm not sure how to reword either one to be generic. These are attributes that appear on some process metrics, and that's probably it. How should this be reworded?

File: model/registry/process.yaml
Comment (braydonk) [reply model/registry/process.yaml:86]: Is it possible to add a brief with a ref? In that case I will just delete it here and add the briefs in with the refs in `process-metrics.yaml`

**Comment by trisch-me** [comment]
I think for this request would be better to create a changelog entry

File: model/registry/process.yaml
Comment (trisch-me) [reply model/registry/process.yaml:79]: That was a temp fix because of broken build-tools. 
There is an open [PR](https://github.com/open-telemetry/semantic-conventions/pull/702) to revert that change

Comment (trisch-me) [reply model/registry/process.yaml:79]: With above mentioned PR merged you could define new attribute group under top process

File: model/registry/process.yaml
Comment (joaopgrassi) [reply model/registry/process.yaml:83]: Hum.. the whole point of the registry is for re-usability. Then I'm wondering if we really need to move all of these attributes to the registry? 🤔 I'm actually now wondering where the need for this came now. I don't think other metrics have their attributes defined in the registry. Could you clarify maybe?

Comment (braydonk) [reply model/registry/process.yaml:83]: This was requested here: https://github.com/open-telemetry/semantic-conventions/pull/330#discussion_r1419228700

File: model/metrics/process-metrics.yaml
Comment (trisch-me) [inline model/metrics/process-metrics.yaml:2]: I think this part should stay here and we are moving into registry only attributes i.e. `state`

File: model/registry/process.yaml
Comment (jsuereth) [reply model/registry/process.yaml:83]: > Hum.. the whole point of the registry is for re-usability. Then I'm wondering if we really need to move all of these attributes to the registry? 🤔 I'm actually now wondering where the need for this came now. I don't think other metrics have their attributes defined in the registry. Could you clarify maybe?

We are currently requiring all attributes in semconv to be part of the registry.  This is to enable x-signal journeys where something my start as an EVENT then become a METRIC.   

I think your concern is related to https://github.com/open-telemetry/semantic-conventions/issues/394.  Let's fix that issue, but no block metrics moving their attributes to the registry.

File: model/registry/process.yaml
Comment (joaopgrassi) [reply model/registry/process.yaml:79]: Let's try to push to get that one merged, so we can move forward with this one 👍

File: model/registry/process.yaml
Comment (joaopgrassi) [reply model/registry/process.yaml:86]: Yeah, you can override when using `ref`. Like here for example: https://github.com/open-telemetry/semantic-conventions/blob/main/model/trace/http.yaml#L66

But if this is just used in metrics as you mentioned above, then leaving here is fine.

File: model/registry/process.yaml
Comment (joaopgrassi) [reply model/registry/process.yaml:79]: We found out we are still blocked by the resource/group type issue with the build tools. Folks are working on it  hopefully soon we can unblock this

File: model/registry/process.yaml
Comment (joaopgrassi) [reply model/registry/process.yaml:83]: I submitted https://github.com/open-telemetry/semantic-conventions/pull/786 for it. But it is still unclear tbh. If metric attributes are "general", like `cpu.state`, which is used in `process` and `system` metrics. Should this go under a `cpu` namespace? Because here it is under `process.cpu.state` which cannot be used in system metrics.

Comment (braydonk) [reply model/registry/process.yaml:83]: `system.cpu.state` and `process.cpu.state` are intentionally different because despite sharing a name, their enums are distinct from one another.

Comment (ChrsMark) [reply model/registry/process.yaml:83]: Would by any chance make sense to unify them? Related to https://github.com/open-telemetry/semantic-conventions/pull/765#discussion_r1506373063

Comment (ChrsMark) [reply model/registry/process.yaml:83]: Having one's allowed values being a subset of the other one's values makes sense to me. Not sure if this can somehow be "templated" with SemConv's rulings.

**Comment by joaopgrassi** [comment]
Not stale

File: model/registry/process.yaml
Comment (ChrsMark) [reply model/registry/process.yaml:83]: @braydonk I have filed https://github.com/open-telemetry/semantic-conventions/issues/840 for this one.

File: model/registry/process.yaml
Comment (lmolkova) [reply model/registry/process.yaml:79]: It should now be unblocked with #820

**Comment by ChrsMark** [comment]
I guess that's still valid :).

**Comment by trisch-me** [comment]
@braydonk there are a lot of changes in the semconv lately. If you want to proceed with this PR please resolve conflicts

**Comment by lmolkova** [comment]
closing in favor of #988

**Pull request closed without merging**
